src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp

```diff
@@ -796,15 +796,77 @@ bool FlowSolverBase::checkSequentialSolutionIncrements( DomainPartition & GEOS_U
-                                                                getName(), fmt::format( "{:.{}f}", m_sequentialTempChange, 3 ) ) );
+                                                                getName(), GEOS_FMT( "{:.{}f}", m_sequentialTempChange, 3 ) ) );
   }
 
   return (m_sequentialPresChange < m_maxSequentialPresChange) && (m_sequentialTempChange < m_maxSequentialTempChange);
 }
 
+string FlowSolverBase::BCMessage::generateMessage( string_view baseMessage,
+                                                   string_view fieldName, string_view setName )
+{
+  return GEOS_FMT( "{} Check if you have added or applied the appropriate fields to "
+                   "the FieldSpecification component with fieldName=\"{}\" "
+                   "and setNames=\"{}\"\n", baseMessage, fieldName, setName );
+}
+
+string FlowSolverBase::BCMessage::pressureConflict( string_view regionName, string_view subRegionName,
+                                                    string_view setName, string_view fieldName )
+{
+  return generateMessage( GEOS_FMT( "Conflicting pressure boundary conditions on set {}/{}/{}\n",
+                                    regionName, subRegionName, setName ), fieldName, setName );
+}
+
+string FlowSolverBase::BCMessage::temperatureConflict( string_view regionName, string_view subRegionName,
+                                                       string_view setName, string_view fieldName )
+{
+  return generateMessage( GEOS_FMT( "Conflicting temperature boundary conditions on set {}/{}/{}\n",
+                                    regionName, subRegionName, setName ), fieldName, setName );
+}
+
+string FlowSolverBase::BCMessage::missingPressure( string_view regionName, string_view subRegionName,
+                                                   string_view setName, string_view fieldName )
+{
+  return generateMessage( GEOS_FMT( "Pressure boundary condition not prescribed on set {}/{}/{}\n",
+                                    regionName, subRegionName, setName ), fieldName, setName );
+}
+
+string FlowSolverBase::BCMessage::missingTemperature( string_view regionName, string_view subRegionName,
+                                                      string_view setName, string_view fieldName )
+{
+  return generateMessage( GEOS_FMT( "Temperature boundary condition not prescribed on set {}/{}/{}\n",
+                                    regionName, subRegionName, setName ), fieldName, setName );
+}
+
+string FlowSolverBase::BCMessage::conflictingComposition( int comp,
+                                                          string_view regionName, string_view subRegionName,
+                                                          string_view setName, string_view fieldName )
+{
+  return generateMessage( GEOS_FMT( "Conflicting composition[{}] boundary conditions on set {}/{}/{}\n",
+                                    comp, regionName, subRegionName, setName ), fieldName, setName );
+}
```

Comment:
Can we have the actual composition name?

```suggestion
string FlowSolverBase::BCMessage::conflictingComposition( string_view compName,
                                                          string_view regionName, string_view subRegionName,
                                                          string_view setName, string_view fieldName )
{
  return generateMessage( GEOS_FMT( "Conflicting composition[{}] boundary conditions on set {}/{}/{}\n",
                                    compName, regionName, subRegionName, setName ), fieldName, setName );
}
```

Same question for `invalidComponentIndex()` and `inconsistency()`.

Reply:
And we miss a space in `composition[{}]` (also in `inconsistency()`)

Reply:
for invalidComponentIndex() cause it's invalid but for conflicComposition (not inconsistency()) yes